src/iface/mod.rs

```diff
@@ -10,3 +10,21 @@ pub use self::neighbor::Neighbor as Neighbor;
 pub(crate) use self::neighbor::Answer as NeighborAnswer;
 pub use self::neighbor::Cache as NeighborCache;
 pub use self::ethernet::Interface as EthernetInterface;
+pub use self::ethernet::Packet as EthernetPacket;
+
+use Result;
+use wire::{IpRepr, UdpRepr, TcpRepr, Icmpv4Repr};
+use phy::{DeviceCapabilities, ChecksumCapabilities};
+
+pub trait PacketFilter {
+    fn process_udp(&mut self, ip_repr: &IpRepr, repr: &UdpRepr) -> Result<()>;
+    fn process_tcp<'frame>(&mut self, timestamp: u64, ip_repr: &IpRepr, repr: &TcpRepr<'frame>) ->
```

Comment:
This really should not return an `EthernetPacket`. We may not even be running over Ethernet!